### PR TITLE
feat: warn if page endpoint body is not serializable

### PR DESCRIPTION
**Pull request by gtm-nayan**
#4944 #5358 #5545

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

File: packages/kit/src/runtime/server/page/load_node.js [hunk 1]
```diff
@@ -3,9 +3,10 @@ import * as set_cookie_parser from 'set-cookie-parser';
 import { normalize } from '../../load.js';
 import { respond } from '../index.js';
 import { LoadURL, PrerenderingURL, is_root_relative, resolve } from '../../../utils/url.js';
-import { check_method_names, is_pojo, lowercase_keys } from '../utils.js';
+import { check_method_names, lowercase_keys } from '../utils.js';
 import { coalesce_to_error } from '../../../utils/error.js';
 import { domain_matches, path_matches } from './cookie.js';
+import { is_pojo, warn_if_not_serdeable } from './utils.js';
```
Comment (dummdidumm) [inline packages/kit/src/runtime/server/page/load_node.js:9]: TIL that serdeable is an actual word 😄

File: packages/kit/src/runtime/server/page/utils.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/**
+ * @param {any} v
+ */
+export function is_pojo(v) {
```
Comment (dummdidumm) [inline packages/kit/src/runtime/server/page/utils.js:4]: It irks me a little that there's another function with the same name which does something different, and both have no short description so you have to jump into the code and usage locations to know what's this about. 
Also: What's the `TODO` about?

File: packages/kit/src/runtime/server/page/utils.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	// TODO: javascript-isms
+	return Object.getPrototypeOf(v) === Object.prototype;
+}
+
+/** @param {any} v */
+export function is_serializable_primitive(v) {
+	// TODO: javascript-isms
```
Comment (dummdidumm) [inline packages/kit/src/runtime/server/page/utils.js:11]: What's the `TODO` about?

File: packages/kit/src/runtime/server/page/load_node.js (hunk 1 above)
Comment (gtm-nayan) [reply packages/kit/src/runtime/server/page/load_node.js:9]: Serde is common in the rust circles. Guess it hasn't carried over much outside, will change if it's not clear.

**Comment by gtm-nayan** [comment]
Ughh, __SVELTEKIT_DEV__ is throwing a reference error. https://github.com/sveltejs/kit/commit/baaf46e612228137e301eea78d530d5b92b44ff4 sounds like it's the same thing however here it might mean massive amounts of param drilling to do the same.

**Comment by Rich-Harris** [comment]
It occurred to me that since we always need to serialize the data, we can do the checks at serialization time with a custom `replacer` argument to `JSON.stringify`, with very little overhead. I implemented that in a new PR since this one is a little out of date since #5778, so I'll close this — thanks

**Pull request closed without merging**